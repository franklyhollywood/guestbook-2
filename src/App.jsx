import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import Auth from './views/Auth/Auth';
import Home from './views/Home/Home';
import Layout from './views/Layout/Layout';

function App() {
  return (
    <Router>
      <Layout>
        <Switch>
          <Route path="/login">
            <Auth />
          </Route>
          <PrivateRoute path="/">
            <Home />
          </PrivateRoute>
        </Switch>
      </Layout>
    </Router>
  );
}

export default App;
