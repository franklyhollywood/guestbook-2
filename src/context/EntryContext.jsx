import { useContext, createContext, useState } from 'react';

// Create the Context
const EntryContext = createContext();

// Create the EntryProvider component
const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);

  return (
    // Our `value` prop is the value we are providing as the `context` value
    // the `value` prop is the data we want to share globally
    // You would not have access to the `Provider` property without creating the context on line 4
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};

// Create custom hook
const useEntries = () => {
  // custom function prefixed with the word `use`

  // `context` is the value being provided
  const context = useContext(EntryContext); // Using this hook is what makes this a custom hook

  // `context` will be undefined when using this custom hook
  // outside of the provider
  if (context === undefined) {
    throw new Error('useEntries must be used within an EntryContext Provider');
  }

  return context;
};

// Export the provider component and the custom hook
export { EntryProvider, useEntries };
