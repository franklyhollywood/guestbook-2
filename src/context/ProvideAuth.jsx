import { createContext, useState } from 'react';
// import dotenv from 'dotenv';
// dotenv.config();
export const authContext = createContext();

export function ProvideAuth({ children }) {
  const [user, setUser] = useState(null);

  const login = (username, password) => {
    const loginSuccessful =
      username === process.env.AUTH_USERNAME &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser(username);
    return loginSuccessful;
  };

  const logout = () => {
    setUser(null);
  };

  return (
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}
