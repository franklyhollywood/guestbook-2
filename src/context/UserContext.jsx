import { useContext, createContext, useState } from 'react';

// create our context
const UserContext = createContext();

// create a user provider that takes in the children

const UserProvider = ({ children }) => {
  // create our text
  const [user, setUser] = useState('');

  // use our provider context
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

// custom hook
const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error(
      'useUser hook must be called within a UserContext Provider'
    );
  }

  return context;
};

export { UserProvider, useUser };

// Create a context file for your state you want to share
// Wrap the provider around your app
// Use the custom hook inside the components that need it.
