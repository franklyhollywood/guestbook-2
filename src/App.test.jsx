import React from 'react';
import App from './App';
import { UserProvider } from './context/UserContext';
import { EntryProvider } from './context/EntryContext';
import { render } from '@testing-library/react';

it('should render index.jsx', () => {
  const { container } = render(
    <React.StrictMode>
      <EntryProvider>
        <UserProvider>
          <App />
        </UserProvider>
      </EntryProvider>
    </React.StrictMode>
  );
  // const sign = screen.getByText('Sign');
  // expect(sign).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
