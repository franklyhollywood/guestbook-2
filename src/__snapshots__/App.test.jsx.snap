// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render index.jsx 1`] = `
<div>
  <div>
    <header>
      <p>
        Please Sign the Guest book
      </p>
    </header>
    <main>
      <h1
        class="text-black p-5 font-bold text-center"
      >
        Please Sign the Guestbook!
      </h1>
      <form
        class="w-full max-w-sm"
      >
        <div>
          <div>
            <label
              class="labelText"
              for="guestName"
            >
              Guest Name
            </label>
          </div>
          <div>
            <input
              id="guestName"
              placeholder="Guest Name..."
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="md:flex md:items-center mb-6"
        >
          <div
            class="md:w-1/3"
          >
            <label
              class="labelText"
              for="guestEntry"
            >
              Guest Entry
            </label>
          </div>
          <div
            class="md:w-2/3"
          >
            <textarea
              id="guestEntry"
              placeholder="Your Entry!"
            />
          </div>
        </div>
        <div>
          <button
            class="button"
            type="submit"
          >
            Sign üñäÔ∏è
          </button>
          
        </div>
      </form>
      <div />
    </main>
    <div>
      I'm a footer
    </div>
  </div>
</div>
`;
